doc/languages-frameworks/emscripten.section.md

```diff
@@ -36,17 +36,11 @@ A few things to note:
 
 ## Declarative usage {#declarative-usage}
 
-Let's see two different examples from `pkgs/top-level/emscripten-packages.nix`:
-
-* `pkgs.zlib.override`
-* `pkgs.buildEmscriptenPackage`
-
-Both are interesting concepts.
-
-A special requirement of the `pkgs.buildEmscriptenPackage` is the `doCheck = true` is a default meaning that each emscriptenPackage requires a `checkPhase` implemented.
+Building emscripten packages requires some special considerations which developers should be aware of. See [`pkgs/top-level/emscripten-packages.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/emscripten-packages.nix) for details and examples.
 
+* `pkgs.buildEmscriptenPackage` sets `docheck = true` by default, which requires that each emscriptenPackage have a `checkPhase`.
 * Use `export EMCC_DEBUG=2` from within a emscriptenPackage's `phase` to get more detailed debug output what is going wrong.
-* ~/.emscripten cache is requiring us to set `HOME=$TMPDIR` in individual phases. This makes compilation slower but also makes it more deterministic.
+* The ``~/.emscripten`` cache requires us to set `HOME=$TMPDIR` in individual phases. This makes compilation slower but also makes it more deterministic. by default the emscripten cache location is set to a new writable `.emscriptencache` directory via the environment variable `EM_CACHE`, because the systemwide default directory is unwritable in the nix store. If overriding the `configurePhase` while creating an emscripten package, this step will need to be included manually  as well.
```

Comment:
Why is that not done in buildEmscriptenPackage ?

Reply:
I should start by acknowledging that I'm still new enough to nixpkgs internals that the distinction between the emscriptenStdenv and buildEmscriptenPackages, as defined by https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/em-modules/generic/default.nix, is a little difficult for me to totally understand. That said, do you mean that there's a way to force those cache settings to happen even if a new `configurePhase` is defined? I assumed that once a user/derivation starts overriding things then it's their responsibility to recreate all the default workarounds, so that nothing else magical is happening behind the scenes once the user has decided to pull back those abstractions.

Reply:
Maybe we can set `EM_CACHE = "$TMP";` in `mkDerivation`?

Reply:
> That said, do you mean that there's a way to force those cache settings to happen even if a new configurePhase is defined?

There are multiple ways. One would be to defined it in an earlier hook for emscripten that is automatically run.